### PR TITLE
feat: jdbc sink user password

### LINKED ISSUE #19766: bug: error reporting accidentally leaked the secret
### Describe the bug

As title.

Perhaps we should use `getConnection(String url, String user, String password)` rather than putting username and password in url.

### Error message/log

```text
Caused by these errors (recent errors listed first):
1: gRPC request to meta service failed: Internal error
2: failed to validate sink
3: sink cannot pass validation: INVALID_ARGUMENT: failed
to connect to target database: 08001: No suitable driver
found for postgresql://xxxx:xxxx@xxxx:xxxx/xxxx?user=xxxx&password=xxxx
```

### To Reproduce

Create a JDBC Postgres Sink.

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## What's changed and what's your intention?

Fix #19766.

The root cause of #19766 is that JDBC URL is included in many JDBC Driver's error message, which is okay. But we put username and password in that URL, which is not a good practice.

This PR uses `getConnection(jdbcUrl, user, password)` instead of `getConnection(jdbcUrl)`, so that the error message won't show the password.

No breaking change - the old syntax can work as well.

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [x]  My PR needs documentation updates. 

Need to update the docs of JDBC sink to use the new format.

No need to add release note - it's just a minor change.

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

File: java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java [hunk 1]
```diff
@@ -56,7 +56,15 @@ public void validate(
...
-        try (Connection conn = DriverManager.getConnection(jdbcUrl);
+        java.util.Properties info = new java.util.Properties();
+        if (config.getUser() != null) {
+            info.put("user", config.getUser());
+        }
+        if (config.getPassword() != null) {
+            info.put("password", config.getPassword());
+        }
+
+        try (Connection conn = DriverManager.getConnection(jdbcUrl, info);
```
Comment (StrikeW) [inline java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java:67]: I think you can directly use `Connection getConnection(String url, String user, String password)` and it has null check inside.

Comment (fuyufjh) [reply java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java:67]: Yeah, that's my original code (as written in the PR description). Here I use `getConnection(jdbcUrl, info)` is to unify the function call in this `try (...)`,

Comment (StrikeW) [reply java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java:67]: Don’t get it.😂 It seems `Connection conn = DriverManager.getConnection(jdbcUrl, config.getUser(), config.getPassword())`. correct me if I am wrong.

**Review by StrikeW — APPROVED** [review]
Rest lgtm

File: java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java (hunk 1 above)
Comment (fuyufjh) [reply java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java]: Make sense. Fixed.

**Pull request merged**